Projects/Server/Main.cs

```diff
@@ -90,33 +102,75 @@ public static bool IsRunningFromXUnit
 
     public static Assembly ApplicationAssembly { get; set; }
     public static Assembly Assembly { get; set; }
-
-    // Assembly file version
     public static Version Version => new(ThisAssembly.AssemblyFileVersion);
-
     public static Process Process { get; private set; }
-
     public static Thread Thread { get; private set; }
-
     private static long _firstTick;
-
     private static long _tickCount;
 
     // Make this available to unit tests for mocking
     internal static DateTime _now;
-
     public static long TickCount => _tickCount;
-
     public static DateTime Now => _now;
-
     public static long Uptime => TickCount - _firstTick;
-
     private static long _cycleIndex;
-    private static readonly double[] _cyclesPerSecond = new double[128];
-
     public static double CyclesPerSecond => _cyclesPerSecond[_cycleIndex];
 
-    public static double AverageCPS => _cyclesPerSecond.Average();
+    public static double AverageCPS
+    {
+        get
+        {
+            var span = _cyclesPerSecond.AsSpan();
+            if (span.IsEmpty)
+            {
+                return 0.0;
+            }
+            
+            double sum = 0.0;
+            foreach (var sample in span)
+            {
+                sum += sample;
+            }
+                
+            return sum / span.Length;
+        }
+    }
+
+    public readonly struct PerformanceMetrics
+    {
+        public readonly double CurrentCPS;
+        public readonly double AverageCPS;
+        public readonly long Uptime;
+        
+        public PerformanceMetrics(double currentCPS, ReadOnlySpan<double> samples, long uptime)
+        {
+            CurrentCPS = currentCPS;
+            AverageCPS = CalculateAverage(samples);
+            Uptime = uptime;
+        }
+        
+        private static double CalculateAverage(ReadOnlySpan<double> samples)
```

Comment:
If we want to avoid having to do a foreach on calculation (which is a good idea), we should add/subtract the incoming and outgoing sample value from a recorded sum, then calculate the average simply requires doing `sum / count`.